cli/flox/src/commands/init/python.rs

```diff
@@ -0,0 +1,664 @@
+                  poetry lock --quiet
+                fi
+
+                echo "Activating poetry virtual environment"
+                source "$(poetry env info --path)/bin/activate"
+
+                poetry install --quiet"#}
+                .to_string(),
+            ),
+            packages: Some(vec![
+                PackageToInstall {
+                    id: "python3".to_string(),
+                    pkg_path: "python3".to_string(),
+                    version: python_version,
+                    input: None,
+                },
+                PackageToInstall {
+                    id: "poetry".to_string(),
+                    pkg_path: "poetry".to_string(),
+                    version: None,
+                    input: None,
+                },
+            ]),
+        }
+    }
+}
+
+/// Information gathered from a pyproject.toml file
+/// <https://packaging.python.org/en/latest/guides/distributing-packages-using-setuptools/#configuring-setup-py>
+#[derive(Debug)]
+struct PyProject {
+    /// Provided python version
+    ///
+    /// [ProvidedVersion::Compatible] if a version compatible with the requirement
+    /// `project.require-python` in the pyproject.toml was found in the catalogs.
+    ///
+    ///
+    /// [ProvidedVersion::Default] if no compatible version was found, but a default version was found.
+    ///
+    /// [ProvidedVersion::Default::requested] is the version requested in the pyproject.toml
+    ///
+    /// May be semver'ish, e.g. ">=3.6"
+    ///
+    /// <https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#python-requires>
+    ///
+    /// [ProvidedVersion::Default::substitute] is the version found in the catalogs instead
+    ///
+    /// Concrete version, not semver!
+    provided_python_version: ProvidedVersion,
+
+    /// Version of poetry found in the catalog
+    pip_version: String,
+}
+
+impl PyProject {
+    fn detect(path: &Path, flox: &Flox) -> Result<Option<Self>> {
+        let pyproject_toml = path.join("pyproject.toml");
+
+        if !pyproject_toml.exists() {
+            return Ok(None);
+        }
+
+        let content = std::fs::read_to_string(&pyproject_toml)?;
+
+        let toml = toml_edit::Document::from_str(&content)?;
+
+        // unlike in poetry, `project.require-python` does not seem to be required
+        //
+        // TODO: check that this is _not (also)_ a poetry file?
+        let required_python_version = toml
+            .get("project")
+            .and_then(|project| project.get("require-python"))
+            .map(|constraint| constraint.as_str().context("expected a string"))
+            .transpose()?
+            .map(|s| s.to_string());
+
+        let provided_python_version = 'version: {
+            let search_default = || {
+                let version =
+                    get_default_package_if_compatible(vec!["python3".to_string()], None, flox)?
```

Comment:
I mean we have to look at the pip stuff anyway, since once you activate the python venv, you dont use the flox installed pip anyway apparently.
`python -m venv venv` installs a pip{3,3**} to `venv/bin` and puts `venv/bin` on the `PATH`!

Reply:
Should we just not install `python311Packages.pip` at all then?

Reply:
yes that's my thinking